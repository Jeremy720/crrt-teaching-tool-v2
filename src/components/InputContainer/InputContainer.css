.InputContainer {
  width: 100%;
}

.InputContainer-radio {
    display: flex;
    flex-direction: column;
    height: 5rem;
    justify-content: space-evenly;
    margin: 0 0 1rem;
}

.modality-radio-label {
  color: #232323;
  font-weight: 600;
}

