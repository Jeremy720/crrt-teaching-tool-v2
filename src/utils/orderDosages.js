const orderDosages = {
  requiredRanges: {
    "sodium": {
      min: 130,
      max: 150
    },
    "potassium": {
      max: 4
    },
    "chloride": {
      min: 95,
      max: 110
    },
    "bicarbonate": {
      min: 20,
      max: 40
    },
    "calcium": {
      min: 0,
      max: 3
    },
    "magnesium": {
      min: 0,
      max: 2
    },
    "phosphorous": {
      min: 0,
      max: 1
    },
    "grossUltraFiltration": {
      min: 0,
      max: 2000
    },
    "replacementFluidFlowRate": {
      min: 0,
      max: 8
    },
  },
  errorMessages: {
    "sodium": "The hospital pharmacy can only compound sodium between 130 and 150.  Use D5W or 3% normal saline if needed",
    "potassium": "The hospital pharmacy will not compound fluid with a potassium above 4 mg/dL",
    "chloride": "The hospital pharmacy can only compound chloride between 95 and 110",
    "bicarbonate": "The hospital pharmacy can only compound bicarbonate between 20 and 40",
    "calcium": "The hospital pharmacy can only compound calcium between 0 and 3 mEq/L",
    "magnesium": "The hospital pharmacy can only compound magnesium between 0 and 2 mgd/L",
    "phosphorous": "The hospital pharmacy cannot compound fluid with > 1mg/dL of phosphorous, due to precipitation of calcium phosphate.",
    "grossUltraFiltration": "The CRRT machine will not accept ultrafiltration rates above 2,000 mL/hour",
    "replacementFluidFlowRate": "The CRRT machine cannot deliver fluid above 8 L/hr",
  }
}

export default orderDosages;