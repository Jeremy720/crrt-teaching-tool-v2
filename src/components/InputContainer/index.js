import React from "react";
import "./InputContainer.css";
import InputCard from "../../containers/InputCard";

const InputContainer = ({
  type,
  currentInputState,
  handleInputChange,
  dosagesToDisplay,
  radioButtonCategory,
  d5W,
  saline3Percent
}) => {
  const inputCards = dosagesToDisplay.map(dosage => {
    return (
      <InputCard
        type={type}
        radioButtonCategory={radioButtonCategory}
        currentInput={
          type === "number" ? currentInputState[dosage] : currentInputState
        }
        handleInputChange={handleInputChange}
        dosage={dosage}
        dosageErrors={currentInputState.dosageErrors}
        key={dosage}
        currentInputState={currentInputState}
        dosagesToDisplay={dosagesToDisplay}
      />
    );
  });

  const getOtherFluidsClass = () => {
    if (d5W === true || saline3Percent === true) {
      return "input-container-other-fluids-visible";
    } else {
      return "input-container-other-fluids-invisible";
    }
  };

  return (
    <div
      className={
        d5W === false && saline3Percent === false
          ? `InputContainer-${type} ${getOtherFluidsClass()}`
          : `InputContainer-${type}`
      }
    >
      {inputCards}
    </div>
  );
};

export default InputContainer;
