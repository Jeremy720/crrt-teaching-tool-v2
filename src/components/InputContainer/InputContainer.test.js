import React from 'react';
import InputContainer from './';
import { shallow } from 'enzyme';
import orderDosages from '../../utils/orderDosages.js';

describe('InputContainer', () => {
  let wrapper;
  let mockState;

  beforeEach(() => {
    mockState = {
			modality: 'Pre-filter CVVH',
			sodium: 4,
			potassium: 3,
			chloride: 5,
			bicarbonate: 6,
			calcium: 8,
			magnesium: 9,
			phosphorous : 3,
			grossUltraFiltration: 2,
			bloodFlowRate: 2,
			replacementFluidFlowRate: 9,
			saline3Percent: false,
			d5W: false,
			sodiumPhosphate15mmol100ml: false,
			anticoagulation: 'None',
			readyForSubmission: false,
			dosageErrors: ['sodium']
		}

    wrapper = shallow(<InputContainer
						type={'text'} 
						currentInputState={mockState}
						handleInputChange={jest.fn()}
						dosagesToDisplay={orderDosages.replacementFluidDosages}
						radioButtonCategory={null}
					/>);
  });

  it('Should render like snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});