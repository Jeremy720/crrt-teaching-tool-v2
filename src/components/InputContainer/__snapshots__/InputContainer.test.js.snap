// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputContainer Should render like snapshot 1`] = `ShallowWrapper {}`;
